solana/gateway-ts/test/unit/onGatewayTokenChange.test.ts

```diff
@@ -94,6 +104,35 @@ describe("onGatewayTokenChange", () => {
       expect(gatewayTokenHistory.length).to.equal(2);
       expect(gatewayTokenHistory[0].state).to.equal(State.ACTIVE);
       expect(gatewayTokenHistory[1].state).to.equal(State.REVOKED);
+      expect(onGatewayTokenChangeId).to.equal(subscriptionId);
+    });
+  });
+
+  context("when subscribing to gateway token state changes", () => {
+    it("should stop listening to gateway state changes", async () => {
+      const gatewayTokenHistory: GatewayToken[] = [];
+      const subscriptionId = 1234;
+
+      // Add subscription and return subscription id
+      // Stop listening to the subscription by providing the subscription id
+      sandbox.stub(connection, "onAccountChange").returns(subscriptionId);
+
+      const removeAccountChangeStub = sandbox
+        .stub(connection, "removeAccountChangeListener")
+        .resolves();
+
+      // register the gateway token change event listener
+      let onGatewayTokenChangeId = onGatewayTokenChange(
+        connection,
+        gatewayTokenKey,
+        () => {}
+      );
+
+      // remove the token event listener
+      removeAccountChangeListener(connection, onGatewayTokenChangeId);
+
+      // expect removeAccountChangeListener to have been called
+      assert(removeAccountChangeStub.called);
```

Comment:
re expectation syntax: i would prefer we stick to expect() or assert() but not mix both.